bulk_site_creator/views.py

```diff
@@ -19,33 +19,103 @@
 from lti_permissions.decorators import lti_permission_required
 
 from common.utils import (get_canvas_site_template,
+                          get_school_data_for_sis_account_id,
                           get_canvas_site_templates_for_school,
-                          get_term_data_for_school)
-
+                          get_term_data_for_school,
+                          get_department_data_for_school,
+                          get_course_group_data_for_school)
 from .schema import JobRecord
-from .utils import (batch_write_item, generate_task_objects,
+from .utils import (batch_write_item,
+                    generate_task_objects,
                     get_course_instance_query_set,
                     get_course_instances_without_canvas_sites,
                     get_department_name_by_id, get_term_name_by_id)
+from boto3.dynamodb.conditions import Key
+from ulid import ULID
 
 logger = logging.getLogger(__name__)
 
 SDK_CONTEXT = SessionInactivityExpirationRC(**settings.CANVAS_SDK_SETTINGS)
 
+dynamodb = boto3.resource("dynamodb")
+table_name = settings.BULK_COURSE_CREATION.get("site_creator_dynamo_table_name")
+table = dynamodb.Table(table_name)
+
 
 @login_required
 @lti_role_required(const.ADMINISTRATOR)
 @lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
 @require_http_methods(["GET", "POST"])
 def index(request):
     sis_account_id = request.LTI["custom_canvas_account_sis_id"]
-    term_ids, _current_term_id_id = get_term_data_for_school(sis_account_id)
+    school_id = sis_account_id.split(":")[1]
+    school_key = f'SCHOOL#{school_id.upper()}'
+    query_params = {
+        'KeyConditionExpression': Key('pk').eq(school_key),
+        'ScanIndexForward': False,
+    }
+    jobs_for_school = table.query(**query_params)['Items']
+
+    context = {
+        'jobs_for_school': jobs_for_school
+    }
+    return render(request, "bulk_site_creator/index.html", context=context)
+
+
+@login_required
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Your code allows safe HTTP methods (i.e `GET`, `HEAD`, `OPTIONS`) and unsafe HTTP methods (i.e `POST`, `PUT`, `DELETE`) to perform an operation on a web application. This could impact its security.  We recommend that you to use safe http methods while performing read-only operations. If you have to use unsafe HTTP methods, use them only to change the state of an appilication. [Learn more](https://cwe.mitre.org/data/definitions/352.html).